test/integration/sequelize.transaction.test.js

```diff
@@ -1,141 +1,181 @@
 'use strict';
 
+const sinon = require('sinon');
 const chai = require('chai'),
   expect = chai.expect,
   Support = require('./support'),
   Transaction = require('sequelize/lib/transaction'),
   current = Support.sequelize,
   delay = require('delay');
 
-if (current.dialect.supports.transactions) {
+const sequelize = Support.sequelize;
+const dialectName = sequelize.dialect.name;
 
-  describe(Support.getTestDialectTeaser('Sequelize#transaction'), () => {
+describe(Support.getTestDialectTeaser('Sequelize#transaction'), () => {
+  if (!current.dialect.supports.transactions) {
+    return;
+  }
 
-    describe('then', () => {
-      it('gets triggered once a transaction has been successfully committed', async function() {
-        let called = false;
+  const stubs = [];
+
+  afterEach(() => {
+    for (const stub of stubs) {
+      stub.restore();
+    }
+
+    stubs.length = 0;
```

Comment:
Uhm. Well. Hmm... `stubs = []` maybe? 🙈

Reply:
I wanted to avoid using `let` but it's not very important so I can change it if you like :)

In the future I want to use `sinon.restore()` after all tests, which would make this redundant, but right now it breaks a bunch of things

Reply:
I think my future me would appreciate that change :)

Reply:
Alright then, I'll need re-approval though :)